### PR TITLE
Modify docs references to kedro[...] to kedro-datasets[...]

### PR DESCRIPTION
**Pull request by jmholzer**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Resolves [#2101](https://github.com/kedro-org/kedro/issues/2101)

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: docs/source/deployment/databricks.md [hunk 1]
```diff
@@ -172,7 +172,7 @@ In your newly-created notebook, put each of the below code snippets into a separ
 * Install the latest version of Kedro compatible with version `0.18.4`
 
 ```console
-%pip install "kedro[spark.SparkDataSet]~=0.18.4"
+%pip install "kedro-datasets[spark.SparkDataSet]~=1.0.0"
```
Comment (merelcht) [inline docs/source/deployment/databricks.md:175]: This should be 
```suggestion
%pip install "kedro~=0.18.4" "kedro-datasets[spark.SparkDataSet]~=1.0.0"
```

I'm not sure about the position of the quotation marks though..

File: docs/source/deployment/databricks.md [hunk 2]
```diff
@@ -172,7 +172,7 @@ In your newly-created notebook, put each of the below code snippets into a separ
 * Install the latest version of Kedro compatible with version `0.18.4`
```
Comment (merelcht) [inline docs/source/deployment/databricks.md:172]: I would also mention installation of `kedro-datasets` in this sentence

**Comment by ankatiyar** [comment]
There's some places in the docs where we mention `kedro.extras.datasets` or are still importing DataSets from `kedro.extras.datasets` in code snippets. I think those can be changed in this PR as well? For example, [The DataCatalog](https://kedro.readthedocs.io/en/stable/data/data_catalog.html#the-data-catalog), [Spaceflights Tutorial FAQs](https://kedro.readthedocs.io/en/stable/tutorial/spaceflights_tutorial_faqs.html#dataseterror-an-exception-occurred-when-parsing-config-for-dataset) etc.

File: docs/source/tutorial/tutorial_template.md [hunk 3]
```diff
@@ -73,7 +73,7 @@ The dependencies above might be sufficient for some projects, but for this tutor
 Add the following lines to your `src/requirements.txt` file:
 
 ```text
-kedro[pandas.CSVDataSet, pandas.ExcelDataSet, pandas.ParquetDataSet]==0.18.4   # Specify optional Kedro dependencies
+kedro-datasets[pandas.CSVDataSet, pandas.ExcelDataSet, pandas.ParquetDataSet]~=1.0.0  # Specify optional Kedro dependencies
```
Comment (AhdraMeraliQB) [inline docs/source/tutorial/tutorial_template.md:76]: Would this be "Specify optional datasets dependencies"?

File: docs/source/deployment/databricks.md (hunk 1 above)
Comment (jmholzer) [reply docs/source/deployment/databricks.md]: Thanks, good catch. I had assumed the user would have installed Kedro, I think this is true for the changes in `visualise_charts_with_plotly.md`, though not in this case.

I made these changes in [05de014](https://github.com/kedro-org/kedro/pull/2103/commits/05de0148bf4f06b3d54cb5525c900a104cb270aa).

File: docs/source/tutorial/tutorial_template.md (hunk 3 above)
Comment (jmholzer) [reply docs/source/tutorial/tutorial_template.md]: Yes, changed the wording in [d974c6b](https://github.com/kedro-org/kedro/pull/2103/commits/d974c6bee1f2700ef002f874c77287758bc6425e) 🙂.

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Pull request merged**